src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -1053,6 +1053,26 @@ int publish_reserve(const DoutPrefixProvider* dpp,
           << "') apply to event of type: '" << to_string(event_type) << "'"
           << dendl;
 
+      // reload the topic in case it changed since the notification was added
+      const RGWPubSub ps(res.store, res.user_tenant, site);
```

Comment:
during the rebase, i noticed that we're using the wrong tenant name here. this is the requesting user's tenant name, but the notification topic could belong to some other tenant or account. we should use the namespace associated with `topic_cfg.owner` instead

Reply:
good catch, currently the lc & replication events are using the `bucket_tenant` and s3 operations are using the `s->user->get_id().tenant`.
this would have had to change in any case to handle the account ?

Reply:
there are other uses of `res.user_tenant` in v1 paths that look okay to me. all i changed was this one `RGWPubSub::get_topic()` call where we load topic metadata for each 'bucket topic' in the notification configuration. can you spot any other places that need to change?

Reply:
> there are other uses of `res.user_tenant` in v1 paths that look okay to me. all i changed was this one `RGWPubSub::get_topic()` call where we load topic metadata for each 'bucket topic' in the notification configuration. can you spot any other places that need to change?

yep i did a quick search too, looks like this was the only place where the account changes were not handled.
This makes me think, you should definitely add a test case in test_bn.py to actually see the event is generated for an account user. (if you modify [this](https://github.com/ceph/ceph/blob/main/src/test/rgw/bucket_notification/test_bn.py#L3884) to also add the account user and run the check for account user and regular user) this will give a good idea on all the events are generated

Reply:
i added a test case that duplicates `test_ps_s3_persistent_notification_http` but with an account user. i saw that test pass locally with the help of commits from https://github.com/ceph/ceph/pull/55663